workbench-core/authorization/src/dynamicAuthorization/wbcGroupManagementPlugin.ts

```diff
@@ -200,6 +200,9 @@ export class WBCGroupManagementPlugin implements GroupManagementPlugin {
       if (currentStatus === 'delete_pending' && newStatus === 'active') {
         throw new ForbiddenError(`Cannot set group '${groupId}' status to active. It is pending delete.`);
       }
+      if (currentStatus === 'deleted' && newStatus === 'delete_pending') {
+        throw new ForbiddenError(`Cannot set group '${groupId}' status to delete_pending. It was deleted.`);
+      }
```

Comment:
Instead of setting the status to `deleted` we should be clearing out the ddb entry. No reason to clog up the db with useless data

Reply:
That should also mean that possible values from `getGroupStatus` and `setGroupStatus` should be different - `getGroupStatus` can only return `active` and `delete_pending`, `setGroupStatus` can additionally be set to `deleted` ?